src/opendr/perception/object_detection_2d/gem/gem_learner.py

```diff
@@ -0,0 +1,1059 @@
+from opendr.perception.object_detection_2d.gem.algorithm.util.detect import detect
+from opendr.perception.object_detection_2d.gem.algorithm.util.sampler import (RandomSampler, SequentialSampler,
+                                                                              DistributedSamplerWrapper)
+from opendr.perception.object_detection_2d.gem.algorithm.datasets import build_dataset
+from opendr.perception.object_detection_2d.gem.algorithm.engine import evaluate, train_one_epoch
+from opendr.perception.object_detection_2d.gem.algorithm.models import build_model, build_criterion, \
+    build_postprocessors
+
+from opendr.engine.constants import OPENDR_SERVER_URL
+from opendr.engine.data import Image
+from opendr.engine.learners import Learner
+from opendr.engine.datasets import ExternalDataset, DatasetIterator, MappedDatasetIterator
+from opendr.engine.target import CocoBoundingBox, BoundingBoxList
+
+import torchvision.transforms as T
+import numpy as np
+import opendr.perception.object_detection_2d.detr.algorithm.util.misc as utils
+from PIL import Image as im
+
+import zipfile
+
+
+class GemLearner(Learner):
+    def __init__(
+            self,
+            model_config_path=os.path.join(
+                os.path.dirname(os.path.realpath(__file__)),
+                "algorithm/configs/model_config.yaml"
+            ),
+            dataset_config_path=os.path.join(
+                os.path.dirname(os.path.realpath(__file__)),
+                "algorithm/configs/dataset_config.yaml"
+            ),
+            iters=10,
+            dataset=None,
+            lr=1e-4,
+            batch_size=1,
+            optimizer="adamw",
+            backbone="resnet50",
+            checkpoint_after_iter=0,
+            checkpoint_load_iter=0,
+            temp_path="temp",
+            device="cuda",
+            threshold=0.7,
+            num_classes=91,
+            return_segmentations=False,
+    ):
+
+        # Pass the shared parameters on super's constructor so they can get initialized as class attributes
+        super(GemLearner, self).__init__(
+            iters=iters,
+            lr=lr,
+            batch_size=batch_size,
+            optimizer=optimizer,
+            backbone=backbone,
+            checkpoint_after_iter=checkpoint_after_iter,
+            checkpoint_load_iter=checkpoint_load_iter,
+            temp_path=temp_path,
+            device=device,
+            threshold=threshold,
+        )
+
+        # Add arguments to a structure like in the original implementation
+        self.args = utils.load_config(model_config_path)
+        self.datasetargs = utils.load_config(dataset_config_path)
+        self.args.backbone = self.backbone
+        self.args.device = self.device
+        self.args.num_classes = num_classes
+        self.args.dataset_file = "coco"
+
+        self.dataset = dataset
+        if self.dataset is not None:
+            print("True")
```

Comment:
Is this message informative for the end user? I think it should be removed, since we do not print anything similar on other Learners.

Reply:
First of all, thank you for reviewing the code Nikos! This indeed should not be there, thank you for spotting this. I will remove it.